src/Core/Domain/Product/Combination/Command/SetCombinationSuppliersCommand.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Combination\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\CommandHandler\SetCombinationSuppliersHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\ValueObject\CombinationId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\ProductSupplier;
+
+/**
+ * Associates supplier with product combination
+ *
+ * @see SetCombinationSuppliersHandlerInterface
+ */
+class SetCombinationSuppliersCommand
+{
+    /**
+     * @var CombinationId
+     */
+    private $combinationId;
+
+    /**
+     * @var array
+     */
+    private $combinationSuppliers;
+
+    /**
+     * @param int $combinationId
+     * @param ProductSupplier[] $combinationSuppliers
+     */
+    public function __construct(
+        int $combinationId,
+        array $combinationSuppliers
```

Comment:
Watch out, remember [the principles](https://devdocs.prestashop.com/1.7/development/architecture/domain/cqrs/#command-and-commandhandler-principles):

> 3. A Command receives only primitive types on input (int, float, string, bool and array).

If you send a collection of classes, then the command is harder to serialize.

Reply:
This is debatable though...

Reply:
yep that was a mistake, I've homogenized with SetProductSuppliersCommand. It accepts an array and transforms it to object in constructor.